web/packages/teleterm/src/ui/hooks/useLogger.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { useRef } from 'react';
+
+import Logger from 'teleterm/logger';
+
+export function useLogger(name: string) {
+  const loggerRef = useRef<Logger>(null);
+  if (loggerRef.current === null) {
+    loggerRef.current = new Logger(name);
+  }
+
+  return loggerRef.current;
```

Comment:
I think returning the contents of a ref like this should be fine, as long as that return value is not used for rendering. I couldn't find any arguments against this in the docs.